components/data-provider/src/data_key.rs

```diff
@@ -0,0 +1,215 @@
+use std::borrow::Borrow;
+use std::borrow::Cow;
+use std::fmt;
+use std::fmt::Write;
+use tinystr::TinyStr16;
+
+/// A top-level collection of related data keys.
+#[non_exhaustive]
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum DataCategory {
+    Decimal,
+    Plurals,
+    PrivateUse(TinyStr16),
+}
+
+impl DataCategory {
+    /// Gets or builds a string form of this DataCategory.
+    pub fn as_str(&self) -> Cow<'static, str> {
+        match self {
+            DataCategory::Decimal => Cow::Borrowed("decimal"),
+            DataCategory::Plurals => Cow::Borrowed("plurals"),
+            DataCategory::PrivateUse(id) => {
+                let mut result = String::from("x-");
+                result.push_str(id.as_str());
```

Comment:
optional:

```rust
format!("x-{}", id).into()
```

Reply:
I'd like to keep what I have, because `format!` is significantly larger code size.